javascript.go

```diff
@@ -0,0 +1,174 @@
+package flixkit
+
+import (
+	"bytes"
+	"embed"
+	"strings"
+	"text/template"
+	"unicode"
+)
+
+type SimpleParameter struct {
+	Name string
+	JsType string
+    Description string
+    FclType string
+    CadType string
+}
+
+type TemplateData struct {
+    Version     string
+    Parameters []SimpleParameter
+    Title       string
+    Description string
+    Location    string
+    IsScript    bool
+    IsLocalTemplate bool
+}
+
+//go:embed templates/*.tmpl
+var templateFiles embed.FS
+
+func GenerateJavaScript(flix *FlowInteractionTemplate, templateLocation string, isLocal bool) (string, error) {
+    tmpl, err := template.ParseFS(templateFiles, "templates/*.tmpl")
+    if err != nil {
+        return "", err
+    }
+
+    methodName := formatTitle(getMessageValue(flix.Data.Messages, "Request"))
+    description := getDescription(*flix)
+    data := TemplateData{
+        Version: flix.FVersion,
+        Parameters: transformArguments(flix.Data.Arguments),
+        Title: methodName,
+        Description: description,
+        Location: templateLocation,
+        IsScript: flix.IsScript(),
+        IsLocalTemplate: isLocal,
+    }
+
+    var buf bytes.Buffer
+    err = tmpl.Execute(&buf, data)
+    return buf.String(), err    
+}
+
+func getDescription(flix FlowInteractionTemplate) string {
```

Comment:
IMHO this should be a method on the sturct not a method that has the struct as a parameter.